File: Practice/src/exercise/RunningRace.py
def solution(players, callings):
    rank = dict(zip(players, range(len(players))))
    for c in callings:
        i = rank[c]
        rank[c] -= 1
        rank[players[i - 1]] += 1
        players[i] = players[i - 1]
        players[i - 1] = c
    return players

File: Practice/src/exercise/RunningRace.js
function solution(players, callings) {
    const rank = {};
    for (let i = 0; i < players.length; i++) {
        rank[players[i]] = i;
    }
    for (let c of callings) {
        let i = rank[c];
        rank[players[i - 1]] = i;
        rank[c] = i - 1;
        players[i] = players[i - 1];
        players[i - 1] = c;
    }
    return players;
}
